Sources/Target/Linux/Thread.cpp

```diff
@@ -164,6 +164,40 @@ ErrorCode Thread::writeCPUState(Architecture::CPUState const &state) {
       ProcessThreadId(process()->pid(), tid()), info, state);
 }
 
+void Thread::fillStoppointData() {
+  HardwareBreakpointManager *hw_bpm = process()->hardwareBreakpointManager();
+
+  if (hw_bpm) {
+    Address addr;
+    BreakpointManager::Mode mode;
+
+    if (hw_bpm->wasHit(addr, mode)) {
+      if (mode != BreakpointManager::kModeExec) {
+        _stopInfo.watchAddr = addr;
+      }
+
+      switch (static_cast<int>(mode)) {
+      case BreakpointManager::kModeExec:
+        _stopInfo.reason = StopInfo::kReasonBreakpoint;
+        return;
+      case BreakpointManager::kModeWrite:
+        _stopInfo.reason = StopInfo::kReasonWriteWatchpoint;
+        return;
+      case BreakpointManager::kModeRead:
+        _stopInfo.reason = StopInfo::kReasonReadWatchpoint;
+        return;
+      case BreakpointManager::kModeRead | BreakpointManager::kModeWrite:
```

Comment:
This doesn't work - error: case value ‘6’ not in enumerated type ‘ds2::BreakpointManager::Mode’ [-Werror=switch]